web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/CdsProfileFetcher.java

```diff
@@ -67,13 +80,21 @@ public CdsProfileFetcher() {
             .setDefaultRequestConfig(requestConfig)
             .useSystemProperties()
             .build());
-        
+
+        this.policyConfiguration = CdsProfileFetcherPolicy.getInstance();
+        resetService = Executors.newSingleThreadScheduledExecutor(
+                ThreadPoolUtils.createDaemonThreadFactory(CdsProfileFetcher.class, "CdsProfilePurgeService"));
+        long cachePurgeInterval = policyConfiguration.getResetPeriodInMinutes();
+        resetService.scheduleAtFixedRate(new CachePurgingRunnable(), cachePurgeInterval, cachePurgeInterval,
```

Comment:
@dhaval24 I wonder if we need an additional executor for purging the cache. If `fetchProfile` is used frequently, you could store the last time it was called and purge the cache after elapsed time is over the threshold.

Reply:
@littleaj there is a slight confusion due to naming. This should more be resetInterval. I do not purge cache and that is not needed. We only reset the counter. If cache part of `InstrumentationKeyResolver` already has the appId resolved there is no need to retry.